openadapt/record.py

```diff
@@ -618,12 +634,30 @@ def record(
 
     # TODO: discard events until everything is ready
 
+    collect_stats()
+
     try:
         while True:
             time.sleep(1)
     except KeyboardInterrupt:
         terminate_event.set()
 
+    collect_stats()
+
+    stats = snapshots[-1].compare_to(snapshots[-2], 'traceback')
+
+    for stat in stats[:3]:
+        print(
+            "{} new KiB {} total KiB {} new {} total memory blocks: ".format(
+                stat.size_diff / 1024,
+                stat.size / 1024,
+                stat.count_diff,
+                stat.count))
+        for line in stat.traceback.format():
+            print(line)
+
+    tr.print_diff()
```

Comment:
Is there a way to get the value as a string, e.g.:
```
trace_str = tr.get_diff_str()
logger.info(f"trace_str=\n{trace_str}")
```

Reply:
Yes, this is the output now:
![image](https://github.com/MLDSAI/OpenAdapt/assets/88949118/ec43bf49-8ec1-463d-a0ed-cce37e5912bd)